spec/unit/block_score/base_spec.rb

```diff
@@ -14,22 +14,90 @@
       subject(:candidate) { create(:candidate, name_first: 'John') }
       before do
         candidate.name_first = 'Jane'
-        candidate.save
+        result = candidate.save
+        expect(result).to be true
       end
 
       it { is_expected.to be_persisted }
       its(:name_first) { is_expected.to eql 'Jane' }
     end
+
+    context 'when encountering a deleted item' do
+      subject(:candidate) { create(:candidate, name_first: 'John') }
+
+      it 'raises and error when previously deleted' do
+        candidate.delete
+        expect(candidate.save).to be false
+      end
+    end
+  end
+
+  describe '#save!' do
+    subject(:candidate) { create(:candidate, name_first: 'John') }
+    context 'when updating an existing candidate' do
+      it 'successfully saves an alteration' do
+        candidate.name_first = 'Jane'
+        result = candidate.save!
+
+        expect(result).to be true
+        expect(candidate.persisted?).to be true
```

Comment:
A slightly nicer way to write this block of code might be:

```ruby
expect { candidate.save! }.to change(candidate, :persisted?)
  .from(false)
  .to(true)
```

The nice thing about that is that it more clearly demonstrates the primary change in the object's `#persisted?` attribute.